public/components/reindexController.react.js

```diff
@@ -37,10 +40,19 @@ export default class ReindexControllerComponent extends React.Component {
             this.loadContentSource(nextProps.routeParams.id, nextProps.routeParams.environment);
             this.loadReindexHistory(nextProps.routeParams.id, nextProps.routeParams.environment);
             this.loadRunningReindex(nextProps.routeParams.id, nextProps.routeParams.environment);
+            this.loadContentSourceWithId(nextProps.routeParams.id);
             this.setState({editModeOn: false});
         }
     }
 
+    loadContentSourceWithId(id) {
+        ContentSourceService.getContentSourcesWithId(id).then(response => {
+            this.setState({
+                contentSourcesForEnvironments: response.contentSources
+            });
+        });
+    }
+
     loadContentSource(id, environment) {
```

Comment:
Is this function still needed now that you have created the new one to return all content sources for a particular id?